### DISCUSSION TITLE
Feature: examine plugin dependencies for wheel availability

### DISCUSSION THREAD

**Discussion by tlambert03** [chanzuckerberg/napari-hub#331] [Ideas]
Related to [this string of posts](https://github.com/napari/napari/issues/3223#issuecomment-971860651), Robert went through a number of plugins and found that some of them are hard to install, particularly if they depend on packages with c-extensions that don't offer wheels.  This is definitely not surprising, but it _is_ probably somewhat predictable, and can (relatively easily) be figured out

**Proposal**

For each plugin, the hub should download the package on pypi and examine the dependencies, then, for each dependency, determine if a wheel is available on pypi.  If not, they should have a notification next to their install button stating something like "warning: this plugin may depend on C-extensions that are not available for your platform.  Installation may fail."

for example, for napari-autolign:

1. download the sdist: https://files.pythonhosted.org/packages/86/ba/c4bff7a6ea501e9ff7bad4039c2e697238249abb94d5fdffbbee7de6e5ad/napari-autolign-0.0.1a1.tar.gz
2. look at the requires.txt in the `egg-info` folder:
```
napari-plugin-engine>=0.1.4
numpy
connected-components-3d
scikit-image
magicgui
scipy
open3d
transforms3d
probreg
```
3. Follow that chain all the way down, looking for packages that don't offer wheels.  For example, `probreg` there doesn't have a wheel in their files list: https://pypi.org/project/probreg/#files

now... in many cases, if they are pure python, it will be fine not to have a wheel: pip will be able to create one just fine at install... so what you kinda need to know, in addition to whether it has a wheel, is whether it actually uses C extensions.

In the case of `probreg`, it has `pybind11` in it's requirements... which is probably a giveaway that it has a C extension. But, you could also check their `setup.py` to see if they import [`setuptools.Extension`](https://github.com/neka-nat/probreg/blob/ab4122a81ddceabea12adc52447880e4e3bef076/setup.py#L4) ... or if they [use `ext_modules` in their setup() command](https://github.com/neka-nat/probreg/blob/ab4122a81ddceabea12adc52447880e4e3bef076/setup.py#L255). 

Having that will _definitely_ help with the general awareness on the part of end users that don't understand why some napari plugin doesn't work.  There is definitely a risk of napari gaining a reputation of not being easy to create plugins for, but at the same time, plugins developers have to be aware that the dependencies they choose _will_ affect whether people can use their plugin or not.

**Comment by sofroniewn** [comment]
Love this @tlambert03! I think it would be really high value all around!
